src/scripts/main.js

```diff
@@ -1,3 +1,25 @@
 'use strict';
 
-// write code here
+const allLists = document.querySelectorAll('li');
+
+allLists.forEach((item) => {
+  const nestedList = item.querySelector('ul');
+
+  if (nestedList) {
+    const span = document.createElement('span');
+
+    const text = item.firstChild.textContent.trim();
```

Comment:
Ensure that 'item.firstChild' is always a text node. If the list item contains other elements, 'firstChild' might not be the text node you expect. Consider using 'item.childNodes' or 'item.contents' to handle different types of child nodes.